services/app.js

```diff
@@ -171,9 +171,22 @@ module.exports = function() {
 
                         var document = {};
                         document.appId = appId;
-                        document.keys = {};
-                        document.keys.js = _generateKey();
-                        document.keys.master = _generateKey();
+                        document.keys ={}
+                        var  masterObj = {};
+                        var  clientObj = {};
+                        var  arr =[];
+
+                        masterObj['key'] = _generateKey();
+                        masterObj['type'] ='master';
+                        masterObj['name'] = 'master';
```

Comment:
When we find existing names, Pluck the names out, convert that array to lower case and then check.